linode/manager
# chore: [M3-6497] - Remove Google Analytics and clean up custom events

## Description 📝
Final Google Analytics -> Adobe Analytics migration PR with the clean up to remove GA from Cloud Manager. We're finally deprecating GA in our 6/26 release!

TODO:
- [x] Merge corresponding internal PR for removing `REACT_APP_GA_ID` and `REACT_APP_GTM_ID`

## Major Changes 🔄
**List highlighting major changes**
- Renames files from `src/utilities/ga` and `src/utilities/ga.test.ts` to `src/utilities/analytics.ts` and `src/utilities/analytics.test.ts`.
- Renames some functions and comments around the app that mention "GA" to "Analytics" to avoid any future confusion.
- Removes `react-ga` as a package.
- Removes references to `GA_ID` and `GTM_ID` environment variables.
- Removes unused custom events.
- Restores firing of the `sendCreateBucketEvent` when an OBJ storage bucket is created -- we used to track this, but it seems to have gotten dropped when refactoring components. 
- Fixes a bug with the Disk Resize Flow custom event that was causing an event to incorrectly fire every time the error toast displayed, rather than every time the docs link was clicked on the error toast! (This was why we had an inflated count for this event.)

## How to test 🧪
1. **How to setup test environment?**
   - Have the `REACT_APP_GA_ID` set in your local `.env`. (Reach out if you need this.)
   - Have the `REACT_APP_ADOBE_ANALYTICS_URL` set in your local `.env`. (Reach out if you need this.)
   - `yarn up`
2. **How to verify changes?**
   - Refer to `src/utilities/analytics.ts` for custom events to test.
   - Test actions that fire custom events and verify in the browser console that (1) no Google Analytics data is logged and (2) Adobe Analytics data is logged. See #9004 for details instructions on setting up + expected output for Adobe logging.
3. **How to run Unit or E2E tests?**
 ```yarn cy:run```
4. Review the corresponding internal PR (ask me for this) to get rid of our build env vars.
